admin/data/sql/migrations/20190131044243.do.fix-fam-pin-generation-bug.sql

```diff
@@ -0,0 +1,49 @@
+--
+-- Find pupils eligible to generate a Familiarisation PIN
+--
+
+ALTER VIEW [mtc_admin].[vewPupilsEligibleForFamiliarisationPinGeneration] AS (
+  SELECT
+    p.id,
+    p.foreName,
+    p.middleNames,
+    p.lastName,
+    p.dateOfBirth,
+    p.school_id,
+    g.group_id,
+    p.urlSlug
+  FROM
+    [mtc_admin].[pupil] p
+      LEFT JOIN [mtc_admin].[pupilAttendance] pa ON (p.id = pa.pupil_id and pa.isDeleted = 0)
+      LEFT JOIN [mtc_admin].[attendanceCode] ac ON (pa.attendanceCode_id = ac.id )
+      LEFT JOIN  [mtc_admin].[pupilGroup] g ON (g.pupil_id = p.id)
+      INNER JOIN [mtc_admin].[pupilStatus] ps ON (p.pupilStatus_id = ps.id)
+  WHERE
+    -- include all pupils except those who are marked as not taking check because they left school
+    (ac.id IS NULL OR ac.code <> 'LEFTT')
```

Comment:
@jon-shipley what about incorrect registration code? shall these be taken into account as well in a similar manner with the ones who left?

Reply:
@GeorgeChatzigiannis - this is the spec, so no, they shouldn't.